### ISSUE TITLE
belongs_to associations no longer respect collection_paths

### ISSUE THREAD

**Issue by keegangroth** [remiprev/her#513]
The change in version 1.0.3 caused a regression in the handling of paths for the belongs_to association. It accepts an optional path now, but the default value for that option ignores the associated class's collection path. Simple broken example:
```
class Company
  include Her::Model
  include_root_in_json true
  collection_path 'my_companies'
end

class Admin
  include Her::Model
  include_root_in_json true
  belongs_to :company
end
```
in order to make the same thing work with 1.0.3, you would have to write `belongs_to :company, path: "#{Company.collection_path}/:company_id"` in the admin class.

One simple solution is to not provide a default value for the path argument.

**Comment by keegangroth** [comment]
just to be clear, this was caused by #511

**Comment by j1mmie** [comment]
I see this is fixed in #514 - any chance #514 will be merged?

**Comment by zacharywelch** [comment]
Confirmed this is a regression 😅 

Although documentation states

> Options Hash (opts):
:path (Path) — The relative path where to fetch the data (defaults to `/class_name.pluralize/id`)

We were [actually](https://github.com/remiprev/her/blob/fd192875777f11135ec2188a4a9b8346226b9561/lib/her/model/paths.rb#L95) defaulting to `resource_path` or `collection_path`. Go figure 😉

The ideal solution would continue to support the custom `path` option as well as default to `resource_path` or `collection_path` based on the type of association. Keep in mind we also maintain a list of associations on the parent class. Specs for that can be found [here](https://github.com/remiprev/her/blob/master/spec/model/associations_spec.rb#L6)

**Comment by zacharywelch** [comment]
@keegangroth's fix can be found in release [v1.1.0](https://github.com/remiprev/her/releases/tag/v1.1.0). Thanks everyone!
